src/irmin/import.ml

```diff
@@ -51,6 +51,15 @@ module List = struct
       | h :: t -> (aux [@tailcall]) (fun t' -> acc (f h :: t')) t
     in
     aux (fun x -> x) l
+
+  let concat l =
+    let rec aux acc curr l =
+      match (curr, l) with
+      | [], [] -> List.rev acc
+      | [], h :: t -> (aux [@tailcall]) acc h t
```

Comment:
I think it could be a good idea to have a special case fo `[], [l]` (and return `List.rev_append acc l`)